File: backend/src/main/java/com/sunrisedental/service/TreatmentService.java
package com.sunrisedental.service;

import com.sunrisedental.dto.PagedResponse;
import com.sunrisedental.dto.TreatmentRequest;
import com.sunrisedental.dto.TreatmentResponse;

import java.util.List;

/**
 * Service interface for Treatment procedures catalog.
 */
public interface TreatmentService {

    TreatmentResponse create(TreatmentRequest request);

    PagedResponse<TreatmentResponse> getAll(
            String search,
            Boolean active,
            int page,
            int size,
            String sortBy,
            String sortDir);

    TreatmentResponse getById(Long id);

    TreatmentResponse update(Long id, TreatmentRequest request);

    void delete(Long id);

    TreatmentSummary getSummary();

    List<TreatmentResponse> getActiveTreatments();

    record TreatmentSummary(
            long total,
            long active,
            long inactive,
            double averagePrice
    ) {}
}

File: frontend/src/services/treatmentService.js
import api from './api';

/**
 * treatmentService — all API calls for Treatments Catalog.
 */
const treatmentService = {

  getAll({ search = '', active = '', page = 0, size = 10, sortBy = 'createdAt', sortDir = 'desc' } = {}) {
    const params = { search, page, size, sortBy, sortDir };
    if (active !== '') params.active = active;

    return api.get('/treatments', { params });
  },

  getById(id) {
    return api.get(`/treatments/${id}`);
  },

  create(data) {
    return api.post('/treatments', data);
  },

  update(id, data) {
    return api.put(`/treatments/${id}`, data);
  },

  delete(id) {
    return api.delete(`/treatments/${id}`);
  },

  getSummary() {
    return api.get('/treatments/summary');
  },

  getActiveTreatments() {
    return api.get('/treatments/active');
  }
};

export default treatmentService;
